src/Orleans/Configuration/ClientConfiguration.cs

```diff
@@ -74,6 +74,13 @@ public enum GatewayProviderType
         public string DataConnectionString { get; set; }
 
         /// <summary>
+        /// the id of the cluster to which this client is connecting. 
+        /// This argument is optional, but is needed in order for the client to receive
+        /// observer messages from different clusters.
+        /// </summary>
+        public string ClusterId { get; set; }
```

Comment:
The problem is that clientID is used in a whole lot of places before the silo is called for the type information in `StartInternal`. It seems difficult to reorganize this.